src/amo/components/AddonMoreInfo/index.js

```diff
@@ -9,33 +9,53 @@ import Link from 'amo/components/Link';
 import { getVersionById, getVersionInfo } from 'core/reducers/versions';
 import { STATS_VIEW } from 'core/constants';
 import translate from 'core/i18n/translate';
+import { fetchCategories } from 'core/reducers/categories';
 import { hasPermission } from 'amo/reducers/users';
-import type { AddonType } from 'core/types/addons';
 import { isAddonAuthor, trimAndAddProtocolToUrl } from 'core/utils';
 import Card from 'ui/components/Card';
 import DefinitionList, { Definition } from 'ui/components/DefinitionList';
 import LoadingText from 'ui/components/LoadingText';
 import { addQueryParams } from 'core/utils/url';
+import type { CategoriesStateType } from 'amo/components/Categories';
 import type { AppState } from 'amo/store';
+import type { ErrorHandlerType } from 'core/errorHandler';
 import type { AddonVersionType, VersionInfoType } from 'core/reducers/versions';
+import type { AddonType } from 'core/types/addons';
 import type { I18nType } from 'core/types/i18n';
+import type { DispatchFunc } from 'core/types/redux';
 import type { ReactRouterLocationType } from 'core/types/router';
 
+import './styles.scss';
+
 type Props = {|
   addon: AddonType | null,
   i18n: I18nType,
 |};
 
 type InternalProps = {|
   ...Props,
+  categoriesState: $PropertyType<CategoriesStateType, 'categories'>,
+  clientApp: string | null,
+  currentVersion: AddonVersionType | null,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
   hasStatsPermission: boolean,
+  loading: boolean,
+  location: ReactRouterLocationType,
   userId: number | null,
-  currentVersion: AddonVersionType | null,
   versionInfo: VersionInfoType | null,
-  location: ReactRouterLocationType,
 |};
 
 export class AddonMoreInfoBase extends React.Component<InternalProps> {
+  constructor(props: InternalProps) {
+    super(props);
+    const { categoriesState, dispatch, errorHandler, loading } = props;
+
+    if (!loading && !categoriesState) {
```

Comment:
We will need logic here to fetch the categories if they are not currently loading and if they haven't already been populated, but rather than look at `categoriesState` you could look at the new prop I suggested, `relatedCategories`.